### PR TITLE
SHOP-6-Collection-Link-Component

### PR DESCRIPTION
**Pull request by sprajams**
- Add simple collection link component
- Configuration for storybook

### CODE REVIEW

File: src/components/CollectionLink/CollectionLink.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            alt={background.alt}
+            className={styles.bgImage}
+          />
+        </div>
+      )}
+    </>
+  );
+  return (
+    // swap outer wrapper between <a> and <div> dependant on if link is present
+    <>
```
Comment (scottqmn) [inline src/components/CollectionLink/CollectionLink.tsx]: Remove fragments. Should be:
```
return href ?
 <Link ...>
: <div ...>

File: src/components/CollectionLink/CollectionLink.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  label: string;
+  background?: { src: string; alt: string };
+}
+
+export const CollectionLink = ({
+  href,
+  label,
+  background,
+}: StorybookDemoProps) => {
+  //main content
```
Comment (scottqmn) [inline src/components/CollectionLink/CollectionLink.tsx]: Remove comments unless describing function utility

File: src/components/CollectionLink/CollectionLink.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        <Link
+          href={href}
+          className={clsx(
+            styles.outer,
+            background && styles.overlay,
+            styles.link
+          )}
+        >
+          {content}
+          <span className={styles.linkArrow}>{`>`}</span>
```
Comment (scottqmn) [inline src/components/CollectionLink/CollectionLink.tsx]: Use corresponding MUI icon

File: src/components/CollectionLink/CollectionLink.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  label,
+  background,
+}: StorybookDemoProps) => {
+  //main content
+  const content = (
+    <>
+      <div>{label}</div>
+      {background && (
+        <div className={styles.bgImageWrap}>
+          {/* disabled warning to use next/image just for this file... */}
```
Comment (scottqmn) [inline src/components/CollectionLink/CollectionLink.tsx]:
```
{/* eslint-disable-next-line @next/next/no-img-element */}

File: src/components/CollectionLink/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,43 @@
+.outer {
+  background: #dddddd;
+  box-shadow: 0 4px 4px rgba(0, 0, 0, 0.25);
```
Comment (scottqmn) [inline src/components/CollectionLink/styles.module.scss]: Remove box-shadow

File: src/components/CollectionLink/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  flex-direction: column;
+  justify-content: space-between;
+
+  font-size: 1.2em;
+  height: 200px;
+  padding: 0.5em 1.5em;
+  position: relative;
+  width: 200px;
+
+  &.overlay {
```
Comment (scottqmn) [inline src/components/CollectionLink/styles.module.scss]: Handle overlay in its own element adjacent to bg

File: src/components/CollectionLink/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,43 @@
+.outer {
+  background: #dddddd;
+  box-shadow: 0 4px 4px rgba(0, 0, 0, 0.25);
+  color: black;
+
+  display: flex;
```
Comment (scottqmn) [inline src/components/CollectionLink/styles.module.scss]: Remove flex, handle chevron positioning with `position: absolute`, set to `display: block` to reset <a> styles

File: src/components/CollectionLink/styles.module.scss [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  padding: 0.5em 1.5em;
+  position: relative;
+  width: 200px;
+
+  &.overlay {
+    color: white;
+    background: rgba(0, 0, 0, 0.5);
+  }
+
+  &.link {
```
Comment (scottqmn) [inline src/components/CollectionLink/styles.module.scss]: Could just leave this styling in the original class

File: src/components/CollectionLink/styles.module.scss [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.bgImageWrap {
+  position: absolute;
+  left: 0;
+  top: 0;
+
+  height: 100%;
+  width: 100%;
+  z-index: -1;
```
Comment (scottqmn) [inline src/components/CollectionLink/styles.module.scss]: Rm

File: src/components/CollectionLink/styles.module.scss [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.bgImage {
+  height: 100%;
+  width: 100%;
+}
+
+.linkArrow {
+  margin-left: auto;
```
Comment (scottqmn) [inline src/components/CollectionLink/styles.module.scss]: Handle with positioning

File: src/components/CollectionLink/CollectionLink.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  },
+} as ComponentMeta<typeof CollectionLink>;
+
+const Template: ComponentStory<typeof CollectionLink> = (args) => (
+  <CollectionLink {...args} />
+);
+
+export const Default = Template.bind({});
+Default.args = {
+  background: {
```
Comment (scottqmn) [inline src/components/CollectionLink/CollectionLink.stories.tsx]: Good usage of placeholder data but can be optimized:
```
const placeholderBg = {
  src: "https://picsum.photos/id/14/200/200",
  alt: "collection 1",
}

... 

Default.args = {
  background: placeholderBg
}

...

NoHref.args = {
  background: placeholderBg
}

**Review by scottqmn — CHANGES_REQUESTED** [review]
Looks good just left some comments with room for cleanup

File: src/components/CollectionLink/CollectionLink.tsx (hunk 3 above)
Comment (sprajams) [reply src/components/CollectionLink/CollectionLink.tsx]: used react-icons instead

**Review by scottqmn — APPROVED** [review]

**Pull request merged**